### PR TITLE
[eigen3] Update to 3.3.8

### PR DESCRIPTION
**Pull request by aluaces**
Update to 3.3.8

Now CMAKEPACKAGE_INSTALL_DIR and PKGCONFIG_INSTALL_DIR are required by Eigen to be relative paths.

- What does your PR fix? Fixes

None.

- Which triplets are supported/not supported? Have you updated the CI baseline?

None added/removed.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

Yes.

### CODE REVIEW

File: ports/eigen3/CONTROL [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Source: eigen3
-Version: 3.3.7
-Port-Version: 7
+Version: 3.3.8
+Port-Version: 1
```
Comment (LilyWangL) [inline ports/eigen3/CONTROL:3]: When port update to new version, `Port-Version` should be removed.

**Review by LilyWangL — CHANGES_REQUESTED** [review]

**Comment by LilyWangL** [comment]
@aluaces Thanks for your PR. Now ports `pcl` `ceres` `caffe2` which depend on `eigen3` build failed with the following error:
```
D:\installed\x86-windows\include\eigen3\Eigen\src/Core/products/Parallelizer.h(162): error C2039: 'eigen_assert_exception': is not a member of 'Eigen'
D:\installed\x86-windows\include\eigen3\Eigen\src/Core/GlobalFunctions.h(54): note: see declaration of 'Eigen'
D:\installed\x86-windows\include\eigen3\Eigen\src/Core/products/GeneralMatrixMatrix.h(487): note: see reference to function template instantiation 'void Eigen::internal::parallelize_gemm<true,GemmFunctor,Eigen::EigenBase<Derived>::Index>(const Functor &,Index,Index,Index,bool)' being compiled
        with
        [
            Derived=Eigen::Transpose<Eigen::Map<const ceres::Matrix,0,Eigen::Stride<0,0>>>,
            Functor=GemmFunctor,
            Index=Eigen::EigenBase<Eigen::Transpose<Eigen::Map<const ceres::Matrix,0,Eigen::Stride<0,0>>>>::Index
        ]
```
Could you continue to fix it? You can download the error log from: https://dev.azure.com/vcpkg/public/_build/results?buildId=44606&view=artifacts&type=publishedArtifacts

File: ports/eigen3/CONTROL (hunk 1 above)
Comment (aluaces) [reply ports/eigen3/CONTROL]: Done.

**Comment by LilyWangL** [comment]
Thanks for your fix. Now `sophus:x64-windows-static` build failed on CI with the following error:
`ninja: error: 'D:/installed/x64-windows-static/debug/lib/openblas.lib', needed by 'test/ceres/test_ceres_se3.exe', missing and no known rule to make it`

**Comment by aluaces** [comment]
Hi, I cannot replicate the build error of `sophus:x64-windows-static` on my computer.  I guess that could be related to different BLAS versions.  On my system I have `libblas.dll.a` and `blas.a`, but not `openblas.lib`, because I have `lapack-reference` installed by default.

I will try to build `openblas` instead, but I wanted to know if the CI is using it.

**Comment by NancyLi1013** [comment]
Hi @aluaces 

Could you please resolve the conflicts?

**Comment by NancyLi1013** [comment]
Ping @aluaces for response, 
is work still being done for this PR?

**Comment by aluaces** [comment]
Yes, I cannot replicate the build error of sophus:x64-windows-static on my computer. I guess that could be related to different BLAS versions. On my system I have libblas.dll.a and blas.a, but not openblas.lib, because I have lapack-reference installed by default.

I will try to build openblas instead, but I wanted to know if the CI is using it.

**Comment by mathisloge** [comment]
@NancyLi1013  isn't this obsolete due to https://github.com/microsoft/vcpkg/pull/15022 ?

**Comment by NancyLi1013** [comment]
@mathisloge 

Thanks for your kindly reminder. `eigen3 `has been updated to 3.3.9 in vcpkg now.

So this PR is not needed any more.

Closing this PR in flavor of #15022.

**Pull request closed without merging**